### PR TITLE
gov: team relationship

### LINKED ISSUE #121: Associate every repository with one and one and only one project
subtask of #118 

Here is the dedicated issue for making decision on member : team : repository mappings.

A team has its reviewer, committer and maintainer, and owns one or more repositories.

The permission of reviewer and committer is similar with current state but indistinguishable among the team. It means a reviewer of the team can review all pull requests in repositories owned by the teams.

The permission of maintainer, based on that of committer, is mainly for exit mechanism and other team level decision making, which will be elaborated later.

I create a pull request #122  with the proposed teams and their members as well as repositories associated with.

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

close #121 

**description**

These files describe relationship in `member : team : repository`. They themselves do not take effect but for textual display.

Since our permissions are flatten into repository level. @hi-rustin and I tend to use GitHub team to maintain the permission and leave the bot simply respect GitHub permission. See also #123 

**issues**

There are two repositories those do not appear in these files: `community`, `rust-prometheus`.

For `community`, its permission is to all maintainers only.

For `rust-prometheus`, its development is with collaboration with guys pay less attention to TiKV and not yet be included in previous SIG model.

I'd like to listen to @BusyJay @breeswish @lucab @mxinden on suggestions. Personally I'm thinking of transfer to prometheus community and retain current permissions.

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
Thank you!
I think it should be `projects/xxxx`. also we might be better off changing it to teams? Because we'll be using GitHub teams to control permissions, so it'll be easier for people to understand?

**Comment by tisonkun** [comment]
Hi @hi-rustin ! Thanks for your comment.

I have no objection on "team" but it is no the term in previous discussion. I'd like to see more feedback.

**Comment by tisonkun** [comment]
For rust-prometheus repository, with a [discussion on Slack](https://tikv-wg.slack.com/archives/C013T4UBP27/p1622794962006600) with @lucab we agree on that it is better to include rust-prometheus as a team (project) under TiKV community.

Then the question is who are the collaborators of that repository so that we can formalize one file to describe it.

@breeswish @lucab @mxinden do you have an idea? I'd try to figure out the collaborators list but it helps if you can say someone who you think should be a committer / maintainer of the repository.

File: project/pd/project.json
Comment (BusyJay) [inline project/pd/project.json:2]: Given that PD and TiKV can't be used alone and PD committers are also actively updating TiKV, I suggest to merge pd project and tikv project.

/cc @nolouch

File: project/raft/project.json
Comment (BusyJay) [inline project/raft/project.json:2]: All maintainers and committers of raft project should also be in tikv project.

Comment (tisonkun) [reply project/raft/project.json:2]: Make sense. Updating...

Comment (tisonkun) [reply project/raft/project.json:2]: Done.

Comment (tisonkun) [reply project/raft/project.json:2]: https://github.com/tikv/community/pull/122/commits/6fff113a394babcdb3f6a753b1eb9ee386146c53

File: project/pd/project.json
Comment (hi-rustin) [inline project/pd/project.json:7]:
```suggestion
    "committers": [
```

File: project/pd/project.json
Comment (hi-rustin) [inline project/pd/project.json:20]:
```suggestion
    "reviewers": [
```

File: project/pd/project.json
Comment (hi-rustin) [inline project/pd/project.json:27]:
```suggestion
    "repositories": [
```

File: project/pd/project.json
Comment (hi-rustin) [inline project/pd/project.json:2]:
```suggestion
    "maintainers": [
```

File: project/raft/project.json
Comment (hi-rustin) [inline project/raft/project.json:1]: ditto

File: project/tikv/project.json
Comment (hi-rustin) [inline project/tikv/project.json:1]: ditto

**Review by hi-rustin — CHANGES_REQUESTED** [review]

File: project/pd/project.json
Comment (disksing) [reply project/pd/project.json]: There are many PD contributors that are not updating TiKV and PD can run with unistore or TiFlash (after it supports direct write) and there are proposals not TiKV related.

Comment (BusyJay) [reply project/pd/project.json]: > There are many PD contributors that are not updating TiKV

You can find that all PD maintainers, committers have updated TiKV except qiuyesuifeng.

> PD can run with unistore or TiFlash (after it supports direct write)

Unfortunately, unistore is not expected to be used in production, and it's purpose is to provide better optimization direction for TiKV. TiFlash is using TiKV to communicate with PD.

The main goal is to have a role that can decide from the global view and choose the design that best suit the evolution of product.

**Review by hi-rustin — COMMENTED** [review]
I still want to use the concept of teams rather than projects.
This will help people understand the new governance structure and associated permissions.

reset LGTM.

File: project/pd/project.json
Comment (JaySon-Huang) [reply project/pd/project.json]: TiFlash is mainly using the Raft layer of TiKV but not the whole TiKV.

IMO, PD provides the ability to schedule data partition (called `Region`) under multi-raft, which is a separate layer from TIKV, a specific key-value storage implementation.

Comment (BusyJay) [reply project/pd/project.json]: > TiFlash is mainly using the Raft layer of TiKV but not the whole TiKV.

This doesn't change the fact that PD is talking to TiKV instead of Tiflash.

> IMO, PD provides the ability to schedule data partition (called Region) under multi-raft...

It's not true. PD's behavior is tightly bound to the behavior of TiKV. Of cause you can simulate a TiKV using the same protocols, but its design is not for general KV storage.

Comment (tisonkun) [reply project/pd/project.json]: @BusyJay @Jayson-Huang @disksing Thanks for your comments. The main issue here might not be software related, which we will discuss on RoadMap and even all its history and future.

The main issue here is about governance. Is PD as a separated project able to create their own proposal and finish development? The original argument I learn from @BusyJay  is that all of PD RFC is coupled with TiKV and most of TiKV proposals contains a "What to do on PD" part.

If we can clarify that

* PD can propose its own (and only) proposal.
* A cross repo proposal can be divided into two or more proposals that can be located in repositories in one project.

Then it should not be a bad direction PD grows its own community.

If current PD members, as individuals, also works on TiKV project repositories, there is no problem those actively contribute to TiKV also reviewer / committer on TiKV project.

File: project/tikv/project.json
Comment (nolouch) [inline project/tikv/project.json:55]:
```suggestion
        "zyguan",
        "nolouch",
        "rleungx",
        "lhy1024"
```

Comment (nolouch) [reply project/tikv/project.json:55]: They were not included in the sig of tikv before, but the contribution from the repo should be enough.

Comment (tisonkun) [reply project/tikv/project.json:55]: OK. I'll push another commit. Be aware that the list is alphabetical sorted. I will add a comment for it also.

**Comment by tisonkun** [comment]
> I still want to use the concept of teams rather than projects.
> This will help people understand the new governance structure and associated permissions.
> 
> reset LGTM.

@hi-rustin I'd suggest you comment on #118 since we define the term there. If there is no objection for at least 72 hours, I'm gonna update all description on weekend.

File: project/prometheus/project.json
Comment (tisonkun) [inline project/prometheus/project.json:13]:
```suggestion
    "committers": [
        "zhouqiang-cl"
    ],
```

File: project/raft/project.json
Comment (tisonkun) [inline project/raft/project.json:15]:
```suggestion
        "overvenus",
        "zhouqiang-cl"
```

**Review by tisonkun — COMMENTED** [review]
Proposed to promote @zhouqiang-cl  as a committer according to his work on release these projects. Also the current situation is he already has write permission for these repository to push release process. We should include it in our gov model.

File: project/pd/project.json
Comment (hi-rustin) [inline project/pd/project.json:2]:
```suggestion
    "_comment": [
```
Why do we need to keep the letter sequence? It doesn't seem to matter if we don't keep it?

I think if we don't have a tool to check it, then it's hard to make sure it stays alphabetical. I would actually prefer to remove the comment and use a tool to make sure that the changes are alphabetical. Maybe it would be better to use eslint or a prettier plugin to check in CI. If you don't have time to add it, then I suggest we remove the comments and open an issue to track it and add it later.

Comment (tisonkun) [reply project/pd/project.json:2]: > Why do we need to keep the letter sequence?

Better to read in general. I don't think it is a strong requirement but we can setup a rule so that others can also propose a fix if it is out of order. See also TiDB document list variables in order and we don't have a linter. I will be glad if there is a linter and agree to open an issue. But let's keep the comment.

Comment (tisonkun) [reply project/pd/project.json:2]: Add a comment https://github.com/tikv/community/issues/121#issuecomment-858235137, I don't think we should create an issue before this pull request merged. It is invalid until then.

File: project/pd/project.json
Comment (tisonkun) [reply project/pd/project.json]: @BusyJay @nolouch It seems PD members believe the project can grow its own community.

@nolouch do you agree that if there is a cross project proposal, said involving TiKV and PD, you'd help it to be divided into two or more proposals each of which can be located in repositories in one project? I think it is the original bad case proposed, that when you create an RFC on PD, it must associated with TiKV and thus @BusyJay supposes they share the same membership.

@BusyJay do you have further concern?

**Comment by tisonkun** [comment]
Here is a large range of cc that ensures all proposed members are notified about this change.

@5kbpers @AndreMouche @BusyJay @Connor1996 @Fullstop000 @HunDunDM @Jibbow @JmPotato @Little-Wallace @Luffbee @MyonKeminta @NingLin-P @TennyZhuang @Xuanwo @Yisaer @accelsao @andylokandy @bobotu @breedwish @breeswish @brson @cfzjywxk @coocood @disksing @ekexium @fredchenbj @gengliqi @hicqu @howardlau1999 @huachaohuang @hunterlxt @imtbkcat @iosmanthus @jackysp @lhy1024 @longfangsong @lonng @lucab @lysu @lzmhhh123 @mxinden @niedhui @nolouch @nrc @overvenus @qiuyesuifeng @rleungx @shafreeck @siddontang @skyzh @sticnarf @sunxiaoguang @tabokie @tiancaiamao @wjhuang2016 @wshwsh12 @xhebox @yiwu-arbug @you06 @youjiali1995 @zhangjinpeng1987 @zhongzc @zhouqiang-cl @zyguan

**Comment by tisonkun** [comment]
... part 2

@sticnarf @sunxiaoguang @tabokie @tiancaiamao @wjhuang2016 @wshwsh12 @xhebox @yiwu-arbug @you06 @youjiali1995 @zhangjinpeng1987 @zhongzc @zhouqiang-cl @zyguan

**Comment by ekexium** [comment]
There are repos that don't use the bot. It seems to me the changes won't make much difference to them. Am I right?

**Comment by tisonkun** [comment]
> There are repos that don't use the bot. It seems to me the changes won't make much difference to them. Am I right?

For the authentication, yes. We will use GitHub permission. For the permission group, it may be affected. See also https://github.com/tikv/community/issues/123#issue-911076933 and especially

> And for repositories where bot is not configured (bot has a limit on events received), the model retains.

@ekexium An example is that you are one of TiKV project committer and have write permission to its repository client-rust. Don't worry.

**Comment by xhebox** [comment]
No objections. Reasonable to me.

**Comment by Xuanwo** [comment]
LGTM

File: project/tikv/project.json
Comment (tiancaiamao) [inline project/tikv/project.json:41]: Please remove me, I don't write or review rust code nowadays.

Comment (tisonkun) [reply project/tikv/project.json:41]: Thanks for your report. I will add an emeritus committer field and move you there. An emeritus committer won't be requested or have effective permission on the repo, but we do respect your works on this project :)

File: project/pd/project.json
Comment (tisonkun) [reply project/pd/project.json]: ping @hi-rustin

File: project/pd/project.json
Comment (tisonkun) [reply project/pd/project.json]: ping @BusyJay @nolouch

File: project/tikv/project.json
Comment (Connor1996) [inline project/tikv/project.json:6]: just curious how the maintainers are decided.

Comment (tisonkun) [reply project/tikv/project.json:6]: According to https://github.com/tikv/community/issues/118 it writes

> Where do current things go?
> Here is the migration details from current SIG membership to project membership.
> migration
> sig-scheduling migrates to PD project
> sig-raft migrates to Raft project
> sig-engine / sig-coprocessor / sig-transaction migrate to TiKV project
> techLeaders and coLeaders migrate to maintainers 

and later @BusyJay proposed https://github.com/tikv/community/pull/122#discussion_r646342537

File: project/tikv/project.json
Comment (bobotu) [inline project/tikv/project.json:24]: Please remove me

Comment (tisonkun) [reply project/tikv/project.json:24]: Thanks for your report. I will add you as an emeritus committer. An emeritus committer won't be requested in code affairs or have effective permission on the repo, but we do respect your works on this project :)

File: project/pd/project.json
Comment (hi-rustin) [reply project/pd/project.json]: OK. Got it. Thanks!

File: project/pd/project.json
Comment (nolouch) [reply project/pd/project.json]: I think the proposal does not necessarily require separation, I would like to make one RFC for one feature, I think put them in one org is accepted. such as `tikv/rfcs` already has [0013-schedule-limit.md](https://github.com/tikv/rfcs/blob/master/text/0013-schedule-limit.md) it only relative PD, and  [0054-joint-consensus.md](https://github.com/tikv/rfcs/blob/master/text/0054-joint-consensus.md) relative two project.

Comment (tisonkun) [reply project/pd/project.json]: @nolouch I think I get your point. We can label an RFC with related projects and require at lease X (maintainer|committer) approval of each project to get the RFC merged.

File: team/pd/team.json
Comment (hi-rustin) [inline team/pd/team.json:1]: I suggest updating the path to `teams/pd/team.json`

**Review by hi-rustin — APPROVED** [review]
Thanks!

LGTM.

File: team/pd/team.json
Comment (tisonkun) [reply team/pd/team.json]: updated

**Comment by hi-rustin** [comment]
/merge

/hold

If you think we're ready to merge it now, please feel free to `/unhold`.

**Review by nolouch — APPROVED** [review]
lgtm

File: project/pd/project.json
Comment (tisonkun) [reply project/pd/project.json]: Ping @BusyJay we may think it as consensus reached to leave TiKV & PD team as is if there is no more objection. We can discuss later on the description how to accept an RFC. Roughly we tag RFC with team label and require X committers of each team give LGTM.

Comment (BusyJay) [reply project/pd/project.json]: LGTM

File: teams/tikv/team.json
Comment (BusyJay) [inline teams/tikv/team.json:6]: @innerr should also be included as maintainer.

Comment (tisonkun) [reply teams/tikv/team.json:6]: Definitely, updated.

File: teams/tikv/team.json
Comment (tisonkun) [inline teams/tikv/team.json:10]:
```suggestion
        "hicqu",
        "innerr",
```

**Comment by tisonkun** [comment]
Thanks to @hi-rustin 's excellent work https://github.com/ti-community-infra/tichi/pull/597 and with the consensus reached at here we are going to apply the change of community governance in days.

File: teams/raft/team.json
Comment (tisonkun) [inline teams/raft/team.json:21]: @BusyJay do you know who is @accelsao? It seems GitHub ID changes and I cannot find the user.

Comment (tisonkun) [reply teams/raft/team.json:21]: OK. It turns to be @jayzhan211

Comment (BusyJay) [reply teams/raft/team.json:21]: Bad idea for github to allow changing ID.

Comment (tisonkun) [reply teams/raft/team.json:21]:
```suggestion
        "jayzhan211"
```

Comment (hi-rustin) [reply teams/raft/team.json:21]: GitHub also has real digital IDs. A better solution is to use real IDs. It's available through the API, but it requires us to make some automated tools to help people get it automatically, otherwise the experience would be terrible.

Comment (hi-rustin) [reply teams/raft/team.json:21]: For example: https://github.com/rust-lang/team/blob/master/people/hi-rustin.toml

**Comment by tisonkun** [comment]
Hi audience of this thread. We are going to apply the authentication rules proposed in #123 with https://github.com/ti-community-infra/configs/pull/358 . Please be aware and report on #123 if you found any bad case.

**Pull request merged**
